### ISSUE TITLE
why do registry lookups on every call?

### ISSUE THREAD

**Issue by dckc** [fabcotech/rchain-multisig#2]
Why not hoist these up and do them once and share the results across calls?

https://github.com/fabcotech/rchain-multisig/blob/ae4c137f6b980de252339b1493626a77a6d5fd94/rholang/multisig.rho#L74-L76

**Comment by fabcotech** [comment]
Good point, it depends on how many time we need to do this lookup, I like to avoid having too many indentations. Currently `registryLookup!(`rho:rchain:revVault`, *ch1) |` is called only two times in the whole file. So I would say we do not need to turn it as a global / called once process.

**Comment by fabcotech** [comment]
Also maybe it consumes lots of gas, that would be an additional argument in favor of turning it into a global process.

**Comment by dckc** [comment]
Right... the phlo / gas cost is the main reason.
